tests/testthat/test-mod_listing.R

```diff
@@ -545,3 +552,112 @@ test_that("mock_table_mm() displays selected dataset after activating global fil
 
   testthat::expect_equal(actual, expected = selected)
 }) # integration
+
+
+test_that("mock_table_mm() displays selected dataset after activating global filter", {
+  # Initialize test app
+  app <- shinytest2::AppDriver$new(
+    app_dir = app_dir, name = "test_global_filter_selected_dataset"
+  )
+
+  selected <- "adae"
+  # Switch dataset
+  app$set_inputs(`multi-dataset` = selected)
+  app$wait_for_idle()
+
+  # Activate global filter
+  app$set_inputs(`global_filter-vars` = "RACE")
+  app$wait_for_idle()
+
+  actual <- app$get_value(input = "multi-dataset")
+
+  # Kill test app
+  app$stop()
+
+  testthat::expect_equal(actual, expected = selected)
+}) # integration
+
+app_dir <- "./apps/listings_app" # applies for all tests within this describe()
+app <- shinytest2::AppDriver$new(app_dir = app_dir, name = "test_listings_app")
+app_dir <- app$get_url()
+
+test_that("Check select all columns works correctly", {
```

Comment:
```suggestion
test_that("selecting all columns works correctly", {
```

Reply:
To keep it consistent: The other test names are designed that they make sense together with test_that(), i.e. "test that selecting all columns works correctly"